s390x Package RPM are Failing

### What happened?

Minor issue on s390x packaging builds.

```
   Compiling conmonrs v0.1.0 (/builddir/build/BUILD/conmon-rs/conmon-rs/server)
error[E0308]: mismatched types
  --> conmon-rs/server/src/oom_watcher.rs:19:48
   |
19 | pub const CGROUP2_SUPER_MAGIC: FsType = FsType(libc::CGROUP2_SUPER_MAGIC as i64);
   |                                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `u32`, found `i64`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `conmonrs` due to previous error
make: *** [Makefile:17: release] Error 101
error: Bad exit status from /var/tmp/rpm-tmp.iFW7as (%build)
```

### What did you expect to happen?

I expect s390x packages to build correctly.

### How can we reproduce it (as minimally and precisely as possible)?

N/A

### Anything else we need to know?

_No response_

### conmon-rs version

None
<details>

```console
$ conmonrs --version
# paste output here
```

</details>


### OS version

None
<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
$ uname -a
# paste output here
```

</details>


### Additional environment details (AWS, VirtualBox, physical, etc.)

None
<details>

</details>

--- a/conmon-rs/server/src/oom_watcher.rs
+++ b/conmon-rs/server/src/oom_watcher.rs
@@ -15,7 +15,16 @@ use tracing::{debug, debug_span, error, Instrument};
 
 #[cfg(any(all(target_os = "linux", target_env = "musl")))]
 pub const CGROUP2_SUPER_MAGIC: FsType = FsType(libc::CGROUP2_SUPER_MAGIC as u64);
-#[cfg(any(all(target_os = "linux", not(target_env = "musl"))))]
+#[cfg(any(all(target_os = "linux", target_arch = "s390x", not(target_env = "musl"))))]
+pub const CGROUP2_SUPER_MAGIC: FsType = FsType(libc::CGROUP2_SUPER_MAGIC as u32);
+#[cfg(any(all(target_os = "linux", target_arch = "armhfp", not(target_env = "musl"))))]
+pub const CGROUP2_SUPER_MAGIC: FsType = FsType(libc::CGROUP2_SUPER_MAGIC as i32);
+#[cfg(any(all(
+    target_os = "linux",
+    not(target_arch = "s390x"),
+    not(target_arch = "armhfp"),
+    not(target_env = "musl")
+)))]
 pub const CGROUP2_SUPER_MAGIC: FsType = FsType(libc::CGROUP2_SUPER_MAGIC as i64);
 
 static CGROUP_ROOT: &str = "/sys/fs/cgroup";

